File: pyg_parent/pyg_sellergoods_interface/src/main/java/com/pyg/service/BrandService.java
package com.pyg.service;

import com.pyg.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {

    List<TbBrand> findAll();

    //分页查询
    PageResult findPage(int pageNum, int pageSize);

    void save(TbBrand brand);

    TbBrand findOne(Long id);

    //批量删除
    void delete(Long[] ids);

    //品牌条件查询
    PageResult findPage(TbBrand brand, int pageNum, int pageSize);

    //查询品牌下拉框数据
    List<Map> selectOptionList();

}

File: pyg_parent/pyg_manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    }

    this.findPage = function (pageNum, pageSize) {
        return $http.get(`../brand/findPage.do?pageNum=${pageNum}&pageSize=${pageSize}`);
    }

    this.save = function (entity) {
        return $http.post("../brand/save.do", entity);
    }

    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    }

    this.del = function (selectIds) {
        return $http.get("../brand/delete.do?ids=" + selectIds);
    }

    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post(`../brand/search.do?pageNum=${pageNum}&pageSize=${pageSize}`, searchEntity);
    }


    //查询品牌下拉框数据
    this.selectOptionList = function () {
        return $http.get("../brand/selectOptionList.do");
    }
});